### PR TITLE
4.x: Upgrade Weld to 5.x #5815

### PR DESCRIPTION
**Pull request by jbescos**
Relates to https://github.com/helidon-io/helidon/issues/5815 and https://github.com/helidon-io/helidon/issues/5821 (only the part of upgrading Jersey)

currently one tck is failing. The test expect to fail with the next, but it is not failing. I set it as draft meanwhile I investigate further.
```
ERROR] org.eclipse.microprofile.config.tck.broken.ConfigPropertiesMissingPropertyInjectionTest.arquillianBeforeClass  Time elapsed: 0.041 s  <<< FAILURE!
org.jboss.arquillian.container.spi.client.container.DeploymentException: Deployment failure!
	at io.helidon.microprofile.arquillian.HelidonDeployableContainer.lambda$deploy$0(HelidonDeployableContainer.java:211)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.helidon.microprofile.arquillian.HelidonDeployableContainer.deploy(HelidonDeployableContainer.java:210)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:151)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:118)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.executeOperation(ContainerDeployController.java:239)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deploy(ContainerDeployController.java:118)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createDeploymentContext(ContainerDeploymentContextHandler.java:71)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.container.DeploymentExceptionHandler.verifyExpectedExceptionDuringDeploy(DeploymentExceptionHandler.java:47)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:92)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:77)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachDeployment(ContainerDeployController.java:232)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachManagedDeployment(ContainerDeployController.java:212)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deployManaged(ContainerDeployController.java:77)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:96)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:89)
	at org.jboss.arquillian.testng.Arquillian.arquillianBeforeClass(Arquillian.java:99)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:65)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:381)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:319)
	at org.testng.internal.invokers.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:178)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:122)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:808)
	at org.testng.TestRunner.run(TestRunner.java:603)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
	at org.testng.SuiteRunner.run(SuiteRunner.java:326)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.runSuites(TestNG.java:1092)
	at org.testng.TestNG.run(TestNG.java:1060)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:284)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:119)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: jakarta.enterprise.inject.spi.DeploymentException: Cannot find configuration key customer.nationality for field public java.lang.String org.eclipse.microprofile.config.tck.broken.ConfigPropertiesBeanMissingProperty.nationality
	at io.helidon.microprofile.config.FieldSetter.validate(FieldSetter.java:164)
	at io.helidon.microprofile.config.ConfigBeanDescriptor.lambda$validate$1(ConfigBeanDescriptor.java:111)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at io.helidon.microprofile.config.ConfigBeanDescriptor.validate(ConfigBeanDescriptor.java:111)
	at io.helidon.microprofile.config.ConfigBeanDescriptor.create(ConfigBeanDescriptor.java:106)
	at io.helidon.microprofile.config.ConfigCdiExtension.processAnnotatedType(ConfigCdiExtension.java:131)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:85)
	at org.jboss.weld.injection.MethodInvocationStrategy$SimpleMethodInvocationStrategy.invoke(MethodInvocationStrategy.java:168)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
	at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:123)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
	at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
	at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessAnnotatedType(ContainerLifecycleEvents.java:207)
	at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessAnnotatedType(ContainerLifecycleEvents.java:176)
	at org.jboss.weld.bootstrap.BeanDeployer.processAnnotatedTypes(BeanDeployer.java:166)
	at org.jboss.weld.bootstrap.BeanDeployment.createTypes(BeanDeployment.java:219)
	at org.jboss.weld.bootstrap.WeldStartup.startInitialization(WeldStartup.java:426)
	at org.jboss.weld.bootstrap.WeldBootstrap.startInitialization(WeldBootstrap.java:79)
	at io.helidon.microprofile.cdi.HelidonContainerImpl.init(HelidonContainerImpl.java:204)
	at io.helidon.common.context.Contexts.runInContext(Contexts.java:137)
	at io.helidon.microprofile.cdi.HelidonContainerImpl.initInContext(HelidonContainerImpl.java:135)
	at io.helidon.microprofile.cdi.HelidonContainerImpl.create(HelidonContainerImpl.java:127)
	at io.helidon.microprofile.cdi.BuildTimeInitializer.lambda$createContainer$2(BuildTimeInitializer.java:63)
	at io.helidon.microprofile.cdi.BuildTimeInitializer.accessContainer(BuildTimeInitializer.java:72)
	at io.helidon.microprofile.cdi.BuildTimeInitializer.createContainer(BuildTimeInitializer.java:62)
	at io.helidon.microprofile.cdi.BuildTimeInitializer.lambda$get$0(BuildTimeInitializer.java:46)
	at io.helidon.microprofile.cdi.BuildTimeInitializer.accessContainer(BuildTimeInitializer.java:72)
	at io.helidon.microprofile.cdi.BuildTimeInitializer.get(BuildTimeInitializer.java:44)
	at io.helidon.microprofile.cdi.ContainerInstanceHolder.fromBuildTimeInitializer(ContainerInstanceHolder.java:65)
	at io.helidon.microprofile.cdi.ContainerInstanceHolder.lambda$get$1(ContainerInstanceHolder.java:56)
	at io.helidon.microprofile.cdi.ContainerInstanceHolder.access(ContainerInstanceHolder.java:91)
	at io.helidon.microprofile.cdi.ContainerInstanceHolder.get(ContainerInstanceHolder.java:53)
	at io.helidon.microprofile.cdi.HelidonContainer.instance(HelidonContainer.java:48)
	at io.helidon.microprofile.server.Server$Builder.build(Server.java:180)
	at io.helidon.microprofile.arquillian.ServerRunner.start(ServerRunner.java:70)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.helidon.microprofile.arquillian.HelidonDeployableContainer.startServer(HelidonDeployableContainer.java:317)
	at io.helidon.microprofile.arquillian.HelidonDeployableContainer.deploy(HelidonDeployableContainer.java:195)
	... 88 more
```

### CODE REVIEW

**Comment by jbescos** [comment]
The issue is that test ConfigPropertiesMissingPropertyInjectionTest contains a bean ConfigPropertiesBeanMissingProperty that is not annotated. This is the PR that fixes it: https://github.com/eclipse/microprofile-config/pull/743

The problem 1 is that fix is not in any maven release, so we cannot use it yet.

The problem 2 is that fix contains other changes that makes other test to fail:

```
[ERROR] Tests run: 469, Failures: 1, Errors: 0, Skipped: 192, Time elapsed: 31.365 s <<< FAILURE! - in TestSuite
[ERROR] org.eclipse.microprofile.config.tck.ConverterTest.arquillianBeforeClass  Time elapsed: 0.07 s  <<< FAILURE!
org.jboss.arquillian.container.spi.client.container.DeploymentException: Deployment failure!
	at io.helidon.microprofile.arquillian.HelidonDeployableContainer.lambda$deploy$0(HelidonDeployableContainer.java:211)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.helidon.microprofile.arquillian.HelidonDeployableContainer.deploy(HelidonDeployableContainer.java:210)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:151)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:118)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.executeOperation(ContainerDeployController.java:239)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deploy(ContainerDeployController.java:118)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createDeploymentContext(ContainerDeploymentContextHandler.java:71)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.container.DeploymentExceptionHandler.verifyExpectedExceptionDuringDeploy(DeploymentExceptionHandler.java:47)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:92)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:77)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachDeployment(ContainerDeployController.java:232)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachManagedDeployment(ContainerDeployController.java:212)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deployManaged(ContainerDeployController.java:77)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:96)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:89)
	at org.jboss.arquillian.testng.Arquillian.arquillianBeforeClass(Arquillian.java:99)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:65)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:381)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:319)
	at org.testng.internal.invokers.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:178)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:122)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:808)
	at org.testng.TestRunner.run(TestRunner.java:603)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
	at org.testng.SuiteRunner.run(SuiteRunner.java:326)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.runSuites(TestNG.java:1092)
	at org.testng.TestNG.run(TestNG.java:1060)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:284)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:119)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-001409: Ambiguous dependencies for type InjectingBean with qualifiers @Default
  at injection point [BackedAnnotatedField] @Inject org.eclipse.microprofile.config.tck.ConverterTest.bean
  at org.eclipse.microprofile.config.tck.ConverterTest.bean(ConverterTest.java:0)
  Possible dependencies: 
  - Managed Bean [class org.eclipse.microprofile.config.tck.ConverterTest$InjectingBean] with qualifiers [@Any @Default],
  - Managed Bean [class org.eclipse.microprofile.config.tck.ConverterTest$InjectingBean] with qualifiers [@Any @Default]

	at org.jboss.weld.bootstrap.Validator.validateInjectionPointForDeploymentProblems(Validator.java:375)
	at org.jboss.weld.bootstrap.Validator.validateInjectionPoint(Validator.java:285)
	at org.jboss.weld.bootstrap.Validator.validateGeneralBean(Validator.java:141)
	at org.jboss.weld.bootstrap.Validator.validateRIBean(Validator.java:162)
	at org.jboss.weld.bootstrap.Validator.validateBean(Validator.java:515)
	at org.jboss.weld.bootstrap.ConcurrentValidator$1.doWork(ConcurrentValidator.java:64)
	at org.jboss.weld.bootstrap.ConcurrentValidator$1.doWork(ConcurrentValidator.java:62)
	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:62)
	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:55)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedInterruptibleCallable.exec(ForkJoinTask.java:1489)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1311)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1840)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1806)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)

```

I think it makes more sense that I run the tests against master of https://github.com/eclipse/microprofile-config and see what are the errors. In case there are errors, I will try to fix them, either in helidon or in microprofile-config. If the fixes applies in microprofile-config, we will wait for a new release.

**Comment by jbescos** [comment]
When running the TCKs to master [microprofile-config](https://github.com/eclipse/microprofile-config), it fails because of this PR: https://github.com/eclipse/microprofile-config/pull/701

Reverting that, it works. I am going to investigate that.

**Comment by jbescos** [comment]
I have created a [PR](https://github.com/eclipse/microprofile-config/pull/767) in microprofile-confiig

**Comment by jbescos** [comment]
In summary, there are 3 tck issues:
- Config-TCK: [This](https://github.com/eclipse/microprofile-config/pull/767) needs to be merged
- Faul-Tolerance-TCK: [This](https://github.com/eclipse/microprofile-fault-tolerance/pull/614) needs to be merged
- LRA-TCK: There is no open PR, but the test that is commented is also failing in main randomly. I decided to exclude it from the test suite because issue was happening before this PR.

File: microprofile/weld/weld-core-impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java [hunk 1]
```diff
@@ -243,8 +244,8 @@ static String getProxyName(String contextId, Class<?> proxiedBeanType, Set<? ext
                 return createProxyName(bean, typeInfo);
             } else {
                 //interface only bean.
-                final StringBuilder name = new StringBuilder();
-                holder = createCompoundProxyName(contextId, bean, typeInfo, name);
+            final StringBuilder name = new StringBuilder();
```
Comment (tomas-langer) [inline microprofile/weld/weld-core-impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java:247]: wrong formatting

**Review by tomas-langer — CHANGES_REQUESTED** [review]
Please create follow up issues for the disabled tests.

File: microprofile/weld/weld-core-impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java (hunk 1 above)
Comment (jbescos) [reply microprofile/weld/weld-core-impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java:247]: It looks we are explicitly throwing it:

```
[ERROR] org.eclipse.microprofile.config.tck.PropertyExpressionsTest.noExpressionComposedButOptional  Time elapsed: 0.001 s  <<< FAILURE!
java.util.NoSuchElementException: Property compose used in expression ${my.prop${compose}} does not exist
	at io.helidon.config.mp.MpConfigImpl.lambda$format$14(MpConfigImpl.java:420)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.helidon.config.mp.MpConfigImpl.format(MpConfigImpl.java:420)
	at io.helidon.config.mp.MpConfigImpl.processExpressions(MpConfigImpl.java:398)
	at io.helidon.config.mp.MpConfigImpl.resolveReferences(MpConfigImpl.java:378)
	at io.helidon.config.mp.MpConfigImpl.lambda$findConfigValue$12(MpConfigImpl.java:339)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.helidon.config.mp.MpConfigImpl.findConfigValue(MpConfigImpl.java:339)
	at io.helidon.config.mp.MpConfigImpl.optionalValue(MpConfigImpl.java:190)
	at io.helidon.config.mp.MpConfigImpl.getOptionalValue(MpConfigImpl.java:130)
	at org.eclipse.microprofile.config.tck.PropertyExpressionsTest.noExpressionComposedButOptional(PropertyExpressionsTest.java:160)
```

File: config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java [hunk 2]
```diff
@@ -335,9 +335,14 @@ private Optional<ConfigValue> findConfigValue(String propertyName) {
...
             }
             String rawValue = value;
-            return applyFilters(propertyName, value)
-                    .map(it -> resolveReferences(propertyName, it))
-                    .map(it -> new ConfigValueImpl(propertyName, it, rawValue, source.getName(), source.getOrdinal()));
+            try {
+                return applyFilters(propertyName, value)
+                        .map(it -> resolveReferences(propertyName, it))
+                        .map(it -> new ConfigValueImpl(propertyName, it, rawValue, source.getName(), source.getOrdinal()));
+            } catch (NoSuchElementException e) {
```
[the remark this answers is not in the document]
Comment (jbescos) [reply config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java:342]: It looks we are explicitly throwing it:

```
[ERROR] org.eclipse.microprofile.config.tck.PropertyExpressionsTest.noExpressionComposedButOptional  Time elapsed: 0.001 s  <<< FAILURE!
java.util.NoSuchElementException: Property compose used in expression ${my.prop${compose}} does not exist
	at io.helidon.config.mp.MpConfigImpl.lambda$format$14(MpConfigImpl.java:420)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.helidon.config.mp.MpConfigImpl.format(MpConfigImpl.java:420)
	at io.helidon.config.mp.MpConfigImpl.processExpressions(MpConfigImpl.java:398)
	at io.helidon.config.mp.MpConfigImpl.resolveReferences(MpConfigImpl.java:378)
	at io.helidon.config.mp.MpConfigImpl.lambda$findConfigValue$12(MpConfigImpl.java:339)
	at java.base/java.util.Optional.map(Optional.java:260)
	at io.helidon.config.mp.MpConfigImpl.findConfigValue(MpConfigImpl.java:339)
	at io.helidon.config.mp.MpConfigImpl.optionalValue(MpConfigImpl.java:190)
	at io.helidon.config.mp.MpConfigImpl.getOptionalValue(MpConfigImpl.java:130)
	at org.eclipse.microprofile.config.tck.PropertyExpressionsTest.noExpressionComposedButOptional(PropertyExpressionsTest.java:160)
```

[the remark this answers is not in the document]
Comment (tomas-langer) [reply config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java:342]: In this case the property exists, but cannot resolve its reference. If the spec says we must throw an exception, then it is correct and the property is not valid.
In this case we cannot return an empty optional (as the property exists). The exception seems to be the correct handling here (again if the spec tells us to fail when a reference cannot be resolved).

[the remark this answers is not in the document]
Comment (tomas-langer) [reply config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java:342]: So there is a sentence in specification that explicitly says that if the property expression cannot be resolved, it should return an empty optional, and there is a new test for that in later versions.

**Comment by tomas-langer** [comment]
@spericas could you please review the fault tolerance part? I have reviewed the rest and I am OK with it.
Thanks

**Pull request merged**